src/shell/dotnet/Shell/MainWindow.xaml.cs

```diff
@@ -106,13 +106,25 @@ public ModuleViewModel(IModuleManifest manifest, ImageSourceProvider imageSource
             }
             else if (manifest.TryGetDetails<NativeManifestDetails>(out var nativeManifestDetails))
             {
-                var icon = System.Drawing.Icon.ExtractAssociatedIcon(Path.GetFullPath(nativeManifestDetails.Path.ToString()));
+                var icon = GetNativeAppIcon(nativeManifestDetails);
+                if (icon != null) { 
+                    ImageSource = System.Windows.Interop.Imaging.CreateBitmapSourceFromHIcon(
+                    icon.Handle,
+                    new Int32Rect { Width = icon.Width, Height = icon.Height },
+                    BitmapSizeOptions.FromEmptyOptions());
+                }
+            }
+        }
 
-                ImageSource = System.Windows.Interop.Imaging.CreateBitmapSourceFromHIcon(
-                icon.Handle,
-                new Int32Rect { Width = icon.Width, Height = icon.Height },
-                BitmapSizeOptions.FromEmptyOptions());
+        public System.Drawing.Icon? GetNativeAppIcon(NativeManifestDetails nativeManifestDetails) {
+
+           var path = Path.GetFullPath(nativeManifestDetails.Path.ToString());
```

Comment:
As we discussed, if a relative path is provided for a (native) application, we will understand that path relative to the manifest file (instead of the working path as it is done here, and in the runner). We think this is easier to understand for the creator of a manifest, more flexible for qa or debugging purposes, and more deterministic in the average use-case.

An example would be the following:
The ComposeUI based app is running from the folder c:\ComposeUIApp
The manifest file is located under c:\Modules\manifest.json

The manifest contains an app with the path .\myApp\myApp.exe

The current implementation would try to find the icon and launch .\ComposeUIApp\myApp\myApp.exe
The expectation instead would be .\Modules\myApp\myApp.exe